### DISCUSSION TITLE
Check if active player before initiating restart/update.

### DISCUSSION THREAD

**Discussion by Zedifuu** [mbround18/valheim-docker#266] [Ideas]
Just an idea to check if there are active players on the server before updating, If players are active skip ? 
Would be nice so that it doesn't update and restart if there are players mid boss/raid/crypt/burial. 

Also like this project, props to rust, planning on moving over from what were currently using! <3

**Comment by anonymus19941** [comment]
Shouldn't be to hard, using the same way as for #297.
This snippet, inserted in [auto_update.sh](https://github.com/mbround18/valheim-docker/blob/main/src/scripts/auto_update.sh), l. 20, would do this (after #314 is fixed):

```bash
    if [ "${PUBLIC:=0}" -eq 0 ]; then
        log "Can't check if players are online when PUBLIC=0"
        log "This is because we cannot query your server via the Steam API"
    else
      export ADDRESS=${ADDRESS:="127.0.0.1:2457"}
      NUMBER_OF_PLAYERS=$(DEBUG_MODE=false odin status --address="${ADDRESS}" --json | jq -r '.players')
      if [ "${NUMBER_OF_PLAYERS}" -gt 0 ]; then
        log "Skipping update, players are online."
        exit 0
      fi
    fi
```

**Reply by mbround18** [reply]
Yeah, this shouldnt be too hard to utilize the same code.

**Reply by mbround18** [reply]
@anonymus19941 take a gander at  https://github.com/mbround18/valheim-docker/pull/315
